ntp-proto/src/packet/extensionfields.rs

```diff
@@ -461,16 +462,17 @@ impl<'a> ExtensionFieldData<'a> {
                     // All previous untrusted fields are now validated
                     this.authenticated.append(&mut this.untrusted);
                 }
-                _ => this
-                    .untrusted
-                    .push(ExtensionField::decode(field).map_err(|e| e.generalize())?),
+                _ => this.untrusted.push(ExtensionField::decode(field)?),
             }
         }
-        if has_invalid_nts {
-            Err(ParsingError::DecryptError((this, size + header_size)))
-        } else {
-            Ok((this, size + header_size, cookie))
-        }
+
+        let result = DeserializedExtensionField {
+            efdata: this,
+            remaining_bytes: &extension_field_bytes[size..],
+            opt_cookie: is_valid_nts.then_some(cookie),
+        };
+
+        Ok(result)
```

Comment:
This reduces an error condition to a success return, which is not ok in my view.